app/main/views.py

```diff
@@ -146,13 +249,13 @@ def pagination_links(pagination, endpoint, args):
     ]
 
 
-def get_json_from_request():
-    if request.content_type != 'application/json':
+def get_json_from_request(key):
```

Comment:
This is almost everywhere called as `get_json_from_request(key)[key]`, maybe it's worth renaming to e.g. `get_request_json_key` and returning `data[key]`